lasagne/layers/sparse/dense.py

```diff
@@ -0,0 +1,24 @@
+import theano.sparse as sparse
+
+from ..dense import DenseLayer as Dense
+
+
+__all__ = [
+    "DenseLayer",
+]
+
+
+class DenseLayer(Dense):
+    def get_output_for(self, input, **kwargs):
+        """
+        Parameters
+        ----------
+        input : theano sparse matrix
+            output from the previous layer
+        """
+        assert type(input) == sparse.basic.SparseVariable
+
+        activation = sparse.basic.dot(input, self.W)
```

Comment:
space dot have many implementation. By default, theano.tensor.dot(a,b) will call theano.sparse.basic.structured_dot().
The main difference is in the grad. Just double check you have the gradient behave like you want.